app/src/org/commcare/android/models/AndroidSessionWrapper.java

```diff
@@ -237,8 +238,13 @@ public void commitStub() throws StorageFullException {
         SecretKey key = CommCareApplication._().createNewSymetricKey();
         
         //TODO: this has two components which can fail. be able to roll them back
-        
-        FormRecord r = new FormRecord("", FormRecord.STATUS_UNSTARTED, getSession().getForm(), key.getEncoded(), null, new Date(0));
+
+        String form = getSession().getForm();
+        // COMMCARE-163637: in landscape mode, this returns null, and the FormRecord should not have any null fields
+        form = form != null ? form : "";
```

Comment:
Do we know why this is returning null in landscape mode? Seems like a big deal for this to have different behavior between the two view styles.